src/app/components/turnos/autocitar/dinamica.component.ts

```diff
@@ -115,7 +117,7 @@ export class DinamicaFormComponent implements OnInit {
     getPrestacionesAgendaDinamicas() {
         let listaPrestaciones = [];
         this.agenda.bloques.forEach(unBloque => {
-            listaPrestaciones = listaPrestaciones.concat(unBloque.tipoPrestaciones);
+            listaPrestaciones = unBloque.tipoPrestaciones.filter(prestacion => this.auth.getPermissions('rup:tipoPrestacion:?').find(permiso => (permiso === prestacion.id)));
```

Comment:
Otra implementacion a considerar, quizas un poquito mas legible.
```suggestion
 const prestacionesProfesional = this.auth.getPermissions('rup:tipoPrestacion:?'); // ids de prestaciones permitidas
        this.agenda.bloques.forEach(unBloque => {
            listaPrestaciones = unBloque.tipoPrestaciones.filter((prestacion: any) => prestacionesProfesional.includes(prestacion.id));
        });
```

Reply:
@negro89 gracias lauchin ahí modifique y agregue la sugerencia